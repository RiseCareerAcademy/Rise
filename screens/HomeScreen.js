import React from "react";
import { Button, Text } from "native-base";
import { Platform, ScrollView, StyleSheet, Image, View } from "react-native";
import { AuthSession, ImagePicker, Permissions } from "expo";
import { MonoText } from "../components/StyledText";
import { connect } from 'react-redux';

export class HomeScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      result: null,
    };

    if (this.props.loggedIn) {
      this.goToMainScreen();
    }
  }

  componentDidUpdate = prevProps => {
    if (prevProps.loggedIn === false && this.props.loggedIn === true) {
      const { navigate } = this.props.navigation;
      navigate('Profile');
    }
  }

  goToMainScreen = () => {
    const { navigate } = this.props.navigation;
    navigate('Main');
  }

  handleMentorPress = async () => {
    // Setup params for Linkedin API
    // For more details: https://developer.linkedin.com/docs/oauth2
    const response_type = 'code';
    const client_id = '7872jtsnbo9n7s';
    const redirectUrl = AuthSession.getRedirectUrl();
    const state =  Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);
    const authUrl =
    `https://www.linkedin.com/oauth/v2/authorization?response_type=${response_type}` +
    `&client_id=${encodeURIComponent(client_id)}` +
    `&redirect_uri=${encodeURIComponent(redirectUrl)}` +
    `&state=${encodeURIComponent(state)}`;

    // Use Expo's AuthSession to connect with the Linkedin API
    // For more details: https://i.expo.io/versions/latest/sdk/auth-session
    const result = await AuthSession.startAsync({ authUrl });
    const { params: { state: responseState, code } } = result;
    let validState = true;
    if (responseState !== state) {
      validState = false;
    }

    // This only displays the results to the screen
    this.setState({ result, authUrl, validState, responseState, state });
  }

  render() {
    const { navigate } = this.props.navigation;
    return (
      <View style={styles.container}>
        <ScrollView
          style={styles.container}
          contentContainerStyle={styles.contentContainer}
        >
          <View style={styles.center}>
            <Image
              source={require("../assets/images/Rise.png")}
              style={{ height: 200, width: 200 }}
            />
          </View>
            <Button full dark onPress={this.handleMentorPress}>
              <Text>Mentor</Text>
            </Button>

          <View>
            <Button full light onPress={() =>
              navigate('Student')
            }>
              <Text style={styles.greyText}>Student</Text>
            </Button>
          </View>
          <View style={styles.center}>
            <Text>
            --- Or ---
            </Text>
          </View>
          <View>
          <Button full dark onPress={() =>
            navigate('SignIn')
          }>
            <Text>SIGN IN</Text>
          </Button>
          {this.state.validi !== undefined ? (
          <Text>{JSON.stringify(this.state.validState) + '\nRequestState: ' + this.state.state + '\nResponseState: ' + this.state.responseState}</Text>
          ) : null}
          {this.state.authUrl ? (
          <Text>{JSON.stringify(this.state.authUrl)}</Text>
          ) : null}
          {this.state.result ? (
          <Text>{JSON.stringify(this.state.result)}</Text>
          ) : null}
        </View>
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff"
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  center: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "center"
  },
  buttonStyle: {
    margin: 10
  },
  greyText: {
    color: "grey"
  },
  contentContainer: {
    paddingTop: 30
  },
});

const mapStateToProps = state => ({
  loggedIn: state.user.loggedIn,
});

export default connect(mapStateToProps, {})(HomeScreen);
