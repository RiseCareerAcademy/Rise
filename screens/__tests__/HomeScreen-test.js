import 'react-native';
import React from 'react';
import { HomeScreen } from '../HomeScreen';
import { shallow } from 'enzyme';

// jest.mock('expo');

jest.mock('expo', () => ({
    AuthSession: {
        getRedirectUrl: () => {},
        startAsync: () => ({
            params: {
                state: 'MyLinkedinState',
                code: 'MyLinkedinCode',
            },
        }),
    },
}));

it('renders correctly', () => {
    const props = {
        navigation: {
            navigate: () => {},
        },
    };
  const wrapper = shallow(<HomeScreen {...props} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders with validi set', () => {
    const props = {
        navigation: {
            navigate: () => {},
        },
    };
  const wrapper = shallow(<HomeScreen {...props} />);
  wrapper.setState({ validi: true });
  expect(wrapper).toMatchSnapshot();
});

it('renders with authUrl set', () => {
    const props = {
        navigation: {
            navigate: () => {},
        },
    };
  const wrapper = shallow(<HomeScreen {...props} />);
  wrapper.setState({ authUrl: 'http://linkedin.com' });
  expect(wrapper).toMatchSnapshot();
});

it('renders with result set', () => {
    const props = {
        navigation: {
            navigate: () => {},
        },
    };
  const wrapper = shallow(<HomeScreen {...props} />);
  wrapper.setState({ result: 'MyResult' });
  expect(wrapper).toMatchSnapshot();
});

it('opens linkedin', async () => {
    const props = {
        navigation: {
            navigate: () => {},
        },
        registerWithLinkedin: jest.fn(),
    };

    const wrapper = shallow(<HomeScreen {...props} />);
    await wrapper.instance().handleMentorPress();
    expect(props.registerWithLinkedin).toBeCalled();
});
