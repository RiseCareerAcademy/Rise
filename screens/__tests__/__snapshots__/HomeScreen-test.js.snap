// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "paddingTop": 30,
      }
    }
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 200,
            "width": 200,
          }
        }
      />
    </Component>
    <Styled(Button)
      dark={true}
      full={true}
      onPress={[Function]}
    >
      <Styled(Text)>
        Mentor
      </Styled(Text)>
    </Styled(Button)>
    <Component>
      <Styled(Button)
        full={true}
        light={true}
        onPress={[Function]}
      >
        <Styled(Text)
          style={
            Object {
              "color": "grey",
            }
          }
        >
          Student
        </Styled(Text)>
      </Styled(Button)>
    </Component>
    <Component
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)>
        --- Or ---
      </Styled(Text)>
    </Component>
    <Component>
      <Styled(Button)
        dark={true}
        full={true}
        onPress={[Function]}
      >
        <Styled(Text)>
          SIGN IN
        </Styled(Text)>
      </Styled(Button)>
    </Component>
  </ScrollViewMock>
</Component>
`;

exports[`renders with authUrl set 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "paddingTop": 30,
      }
    }
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 200,
            "width": 200,
          }
        }
      />
    </Component>
    <Styled(Button)
      dark={true}
      full={true}
      onPress={[Function]}
    >
      <Styled(Text)>
        Mentor
      </Styled(Text)>
    </Styled(Button)>
    <Component>
      <Styled(Button)
        full={true}
        light={true}
        onPress={[Function]}
      >
        <Styled(Text)
          style={
            Object {
              "color": "grey",
            }
          }
        >
          Student
        </Styled(Text)>
      </Styled(Button)>
    </Component>
    <Component
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)>
        --- Or ---
      </Styled(Text)>
    </Component>
    <Component>
      <Styled(Button)
        dark={true}
        full={true}
        onPress={[Function]}
      >
        <Styled(Text)>
          SIGN IN
        </Styled(Text)>
      </Styled(Button)>
      <Styled(Text)>
        "http://linkedin.com"
      </Styled(Text)>
    </Component>
  </ScrollViewMock>
</Component>
`;

exports[`renders with result set 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "paddingTop": 30,
      }
    }
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 200,
            "width": 200,
          }
        }
      />
    </Component>
    <Styled(Button)
      dark={true}
      full={true}
      onPress={[Function]}
    >
      <Styled(Text)>
        Mentor
      </Styled(Text)>
    </Styled(Button)>
    <Component>
      <Styled(Button)
        full={true}
        light={true}
        onPress={[Function]}
      >
        <Styled(Text)
          style={
            Object {
              "color": "grey",
            }
          }
        >
          Student
        </Styled(Text)>
      </Styled(Button)>
    </Component>
    <Component
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)>
        --- Or ---
      </Styled(Text)>
    </Component>
    <Component>
      <Styled(Button)
        dark={true}
        full={true}
        onPress={[Function]}
      >
        <Styled(Text)>
          SIGN IN
        </Styled(Text)>
      </Styled(Button)>
      <Styled(Text)>
        "MyResult"
      </Styled(Text)>
    </Component>
  </ScrollViewMock>
</Component>
`;

exports[`renders with validi set 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "paddingTop": 30,
      }
    }
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 200,
            "width": 200,
          }
        }
      />
    </Component>
    <Styled(Button)
      dark={true}
      full={true}
      onPress={[Function]}
    >
      <Styled(Text)>
        Mentor
      </Styled(Text)>
    </Styled(Button)>
    <Component>
      <Styled(Button)
        full={true}
        light={true}
        onPress={[Function]}
      >
        <Styled(Text)
          style={
            Object {
              "color": "grey",
            }
          }
        >
          Student
        </Styled(Text)>
      </Styled(Button)>
    </Component>
    <Component
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)>
        --- Or ---
      </Styled(Text)>
    </Component>
    <Component>
      <Styled(Button)
        dark={true}
        full={true}
        onPress={[Function]}
      >
        <Styled(Text)>
          SIGN IN
        </Styled(Text)>
      </Styled(Button)>
    </Component>
  </ScrollViewMock>
</Component>
`;
